project/Component/NefJekyll/Models/RenderJekyllEnvironment.swift

```diff
@@ -44,12 +44,16 @@ public struct RenderJekyllEnvironment<A> {
     static func permalink(info: RenderEnvironmentInfo) -> IO<CoreRenderError, String> {
         switch info {
         case let .info(playground, page):
-            return IO.pure("/\(docs)/\(playground.escapedTitle)/\(page.escapedTitle)/")^
+            return IO.pure("/\(docs)/\(pagePathComponent(playground: playground, page: page))/")^
         default:
             return IO.raiseError(.renderEmpty)^
         }
     }
     
+    static func pagePathComponent(playground: RenderingURL, page: RenderingURL) -> String {
+        "\(playground.escapedTitle)/\(page.escapedTitle)"
+    }
```

Comment:
This looks like it does not belong here. I'd say it should be an instance method of `RenderingURL`. Still, not optimal, but you are using this as a static method from the outside, which should not happen.